Source/Mod/Menu/OnScreenKeyboardMenu.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+
+namespace Celeste64.Mod;
+
+public class OnScreenKeyboardMenu : Menu
+{
+	public Target Target;
+	public Target GameTarget;
+
+	private int currentPage = 0;
+	private int currentRow = 0;
+	private int currentColumn = 0;
+
+	private const int rows = 7;
+	private const int columns = 9;
+
+	private int CurrentPageStart => currentPage * columns * rows;
+	private int CurrentIndex => currentRow * columns + currentColumn;
+	bool shiftMode;
+
+	string setText = string.Empty;
+
+	InputField Owner;
```

Comment:
These variables should probably all have explicit access modifiers (i.e. pubic, private, etc...)